models/repo/repo_list.go

```diff
@@ -688,6 +688,24 @@ func FindUserCodeAccessibleRepoIDs(user *user_model.User) ([]int64, error) {
 	return repoIDs, nil
 }
 
+// FindUserCodeAccessibleUserRepoIDs finds all at Code level accessible repositories' ID by the user's id with specific owner.
+func FindUserCodeAccessibleUserRepoIDs(ownerID int64, user *user_model.User) ([]int64, error) {
+	repoIDs := make([]int64, 0, 10)
+	cond := builder.NewCond()
+	cond.And(
+		builder.Eq{"owner_id": ownerID},
+		AccessibleRepositoryCondition(user, unit.TypeCode),
+	)
+	if err := db.GetEngine(db.DefaultContext).
+		Table("repository").
+		Cols("id").
+		Where(cond).
+		Find(&repoIDs); err != nil {
+		return nil, fmt.Errorf("FindUserCodeAccessibleUserRepoIDs: %v", err)
```

Comment:
Either that, or we throw out this "if" and return the error and the result directly

Reply:
agree, did that also for other method that was already there quite simillar